public/app/core/components/TransformersUI/rowsToFields/rowsToFields.ts

```diff
@@ -0,0 +1,193 @@
+import { map } from 'rxjs/operators';
+import {
+  anyToNumber,
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  DataTransformerInfo,
+  Field,
+  FieldColorModeId,
+  FieldConfig,
+  FieldType,
+  getFieldDisplayName,
+  ReducerID,
+  ThresholdsMode,
+} from '@grafana/data';
+
+export interface RowToFieldsTransformOptions {
+  nameField?: string;
+  valueField?: string;
+  mappings?: RowToFieldsTransformMappings[];
+}
+
+export interface RowToFieldsTransformMappings {
+  fieldName: string;
+  reducerId?: ReducerID;
+  configProperty: string;
+}
+
+export const rowsToFieldsTransformer: DataTransformerInfo<RowToFieldsTransformOptions> = {
+  id: DataTransformerID.rowsToFields,
+  name: 'Rows to fields',
+  description: 'Convert rows to fields with dynamic config',
+  defaultOptions: {},
+
+  /**
+   * Return a modified copy of the series.  If the transform is not or should not
+   * be applied, just return the input series
+   */
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        // Ignore if we have more than one frame
+        if (data.length !== 1) {
+          return data;
+        }
+
+        return [rowsToFields(options, data[0])];
```

Comment:
Why only process one frame?  maybe:

```suggestion
        return data.map( (frame) => rowsToFields(options, frame) );
```

Reply:
I guess no reason :)